File: ClientApp/src/app/models/Wiki.ts
import { User } from "./User";

export class Wiki {
  id: number;
  url: string;
  name: string;
  description: string;
  creationDate: string;
  owner: User;
  maintainers: User[];

  constructor(object: any) {
    this.id = object?.id || 0;
    this.url = object?.url || '';
    this.name = object?.name || '';
    this.description = object?.description || '';
    if(object.owner != null){
      this.owner = new User(object.owner);
    }else{
      this.owner = new User({});
    }
    if(object.creationDate != null){
      this.creationDate = object.creationDate
    }
    this.maintainers = [];
    object?.maintainers?.forEach((obj:string) => { this.maintainers.push(new User(obj)); })||[];
  }
}

File: Server/Entities/Wiki/Models/Wiki.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace OpenWiki.Server.Entities {
    public class Wiki {
        [Key]
        public long ID { get; set; }
        public string URL { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime CreationDate { get; set; }

        [ForeignKey("OwnerId")]
        public ApplicationUser Owner { get; set; }
        public ICollection<ApplicationUser> Maintainers { get; set; }

        public bool CanBeModifiedBy(ApplicationUser user) {
            if(Owner == user) {
                return true;
            }
            foreach(ApplicationUser maintainer in Maintainers) {
                if(maintainer == user) {
                    return true;
                }
            }
            return false;
        }
    }
}
